Polars write_database + ADBC PostgreSQL >= v1.3.0 drops Arrow timestamp time zone — expected?

### What would you like help with?

Starting with adbc-driver-postgresql v1.3.0, if I try to write my DF that has a timezone in it, the TZ is dropped.

I think this has to do with this:
https://arrow.apache.org/adbc/current/driver/postgresql.html#timestamp

Specifically this part:
>When binding a timestamp value, the time zone (if present) is ignored. The value will be converted to microseconds and adjusted to the PostgreSQL epoch (2000-01-01) and so may overflow/underflow; an error will be returned if this would be the case.

However, I thought I was using the bulk ingestion feature through polars `write_database` method, which would preserve the TZ.

Quick code snippet:
```
from datetime import datetime

import adbc_driver_postgresql.dbapi

import polars as pl

df = pl.DataFrame({
    "timestamp": pl.datetime_range(
        start=datetime(2025, 5, 28, 8, 0),
        end=datetime(2025, 5, 28, 12, 0),
        interval="1h",
        time_zone="UTC",
        eager=True
    )
})

table_name = "test_table"
uri = "postgresql://mac@localhost:5432/mac"

# Connect using ADBC
with adbc_driver_postgresql.dbapi.connect(uri) as conn:
    with conn.cursor() as cur:
        cur.execute(f"DROP TABLE IF EXISTS {table_name}")
        conn.commit()

df.write_database(
    table_name=table_name,
    connection=uri,
    engine="adbc",
    if_table_exists="replace"
)

schema_query = f"""
SELECT data_type
FROM information_schema.columns 
WHERE table_name = '{table_name}'
"""

df_schema = pl.read_database_uri(schema_query, uri, engine="adbc")
actual_data_type = df_schema["data_type"][0]
assert actual_data_type == "timestamp with time zone", actual_data_type
```

Running:
```
(adbc_test) mac@macbookpro adbc_test % uv pip install adbc-driver-postgresql==1.2.0
Resolved 4 packages in 15ms
Uninstalled 1 package in 1ms
Installed 1 package in 2ms
 - adbc-driver-postgresql==1.3.0
 + adbc-driver-postgresql==1.2.0
(adbc_test) mac@macbookpro adbc_test % python test_script.py                       
(adbc_test) mac@macbookpro adbc_test % uv pip install adbc-driver-postgresql==1.3.0
Resolved 4 packages in 2ms
Uninstalled 1 package in 1ms
Installed 1 package in 2ms
 - adbc-driver-postgresql==1.2.0
 + adbc-driver-postgresql==1.3.0
(adbc_test) mac@macbookpro adbc_test % python test_script.py                       
Traceback (most recent call last):
  File "/Users/mac/adbc_test/test_script.py", line 41, in <module>
    assert actual_data_type == "timestamp with time zone", actual_data_type
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: timestamp without time zone
```

--- a/c/driver/postgresql/postgres_type.h
+++ b/c/driver/postgresql/postgres_type.h
@@ -607,10 +607,10 @@ inline ArrowErrorCode PostgresType::FromSchema(const PostgresTypeResolver& resol
     case NANOARROW_TYPE_INTERVAL_MONTH_DAY_NANO:
       return resolver.Find(resolver.GetOID(PostgresTypeId::kInterval), out, error);
     case NANOARROW_TYPE_TIMESTAMP:
-      if (strcmp("", schema_view.timezone) == 0) {
-        return resolver.Find(resolver.GetOID(PostgresTypeId::kTimestamptz), out, error);
-      } else {
+      if (std::string_view(schema_view.timezone).empty()) {
         return resolver.Find(resolver.GetOID(PostgresTypeId::kTimestamp), out, error);
+      } else {
+        return resolver.Find(resolver.GetOID(PostgresTypeId::kTimestamptz), out, error);
       }
     case NANOARROW_TYPE_DECIMAL128:
     case NANOARROW_TYPE_DECIMAL256:

--- a/c/driver/postgresql/postgres_type_test.cc
+++ b/c/driver/postgresql/postgres_type_test.cc
@@ -304,6 +304,24 @@ TEST(PostgresTypeTest, PostgresTypeFromSchema) {
   schema.reset();
 
   ArrowSchemaInit(schema.get());
+  ASSERT_EQ(ArrowSchemaSetTypeDateTime(schema.get(), NANOARROW_TYPE_TIMESTAMP,
+                                       NANOARROW_TIME_UNIT_MICRO, ""),
+            NANOARROW_OK);
+  EXPECT_EQ(PostgresType::FromSchema(resolver, schema.get(), &type, nullptr),
+            NANOARROW_OK);
+  EXPECT_EQ(type.type_id(), PostgresTypeId::kTimestamp);
+  schema.reset();
+
+  ArrowSchemaInit(schema.get());
+  ASSERT_EQ(ArrowSchemaSetTypeDateTime(schema.get(), NANOARROW_TYPE_TIMESTAMP,
+                                       NANOARROW_TIME_UNIT_MICRO, "America/Phoenix"),
+            NANOARROW_OK);
+  EXPECT_EQ(PostgresType::FromSchema(resolver, schema.get(), &type, nullptr),
+            NANOARROW_OK);
+  EXPECT_EQ(type.type_id(), PostgresTypeId::kTimestamptz);
+  schema.reset();
+
+  ArrowSchemaInit(schema.get());
   ASSERT_EQ(ArrowSchemaSetType(schema.get(), NANOARROW_TYPE_LIST), NANOARROW_OK);
   ASSERT_EQ(ArrowSchemaSetType(schema->children[0], NANOARROW_TYPE_BOOL), NANOARROW_OK);
   EXPECT_EQ(PostgresType::FromSchema(resolver, schema.get(), &type, nullptr),

--- a/python/adbc_driver_postgresql/tests/test_dbapi.py
+++ b/python/adbc_driver_postgresql/tests/test_dbapi.py
@@ -274,6 +274,36 @@ def test_stmt_ingest_multi(postgres: dbapi.Connection) -> None:
         assert cur.fetch_arrow_table() == table
 
 
+def test_stmt_ingest_timestamptz(postgres: dbapi.Connection) -> None:
+    # Regression test for https://github.com/apache/arrow-adbc/issues/2901
+    table = pyarrow.table(
+        [
+            [1],
+            [datetime.datetime(2023, 1, 2, 3, 4, 5, tzinfo=datetime.timezone.utc)],
+        ],
+        names=["ints", "tstz"],
+    )
+
+    with postgres.cursor() as cur:
+        cur.execute("DROP TABLE IF EXISTS test_ingest")
+        # Make sure we aren't in UTC time zone
+        cur.execute("SET TIME ZONE 'Asia/Tokyo'")
+        postgres.commit()
+
+        cur.execute("SELECT current_setting('TIMEZONE')")
+        assert cur.fetchone() == ("Asia/Tokyo",)
+
+        cur.adbc_ingest(
+            "test_ingest",
+            table,
+            mode="create",
+        )
+        postgres.commit()
+
+        cur.execute("SELECT * FROM test_ingest ORDER BY ints")
+        assert cur.fetch_arrow_table() == table
+
+
 def test_ddl(postgres: dbapi.Connection):
     with postgres.cursor() as cur:
         cur.execute("DROP TABLE IF EXISTS test_ddl")

